ArchiSteamFarm/GitHub.cs

```diff
@@ -0,0 +1,220 @@
+//    / \    _ __  ___ | |__  (_)/ ___| | |_  ___   __ _  _ __ ___  |  ___|__ _  _ __  _ __ ___
+//   / _ \  | '__|/ __|| '_ \ | |\___ \ | __|/ _ \ / _` || '_ ` _ \ | |_  / _` || '__|| '_ ` _ \
+//  / ___ \ | |  | (__ | | | || | ___) || |_|  __/| (_| || | | | | ||  _|| (_| || |   | | | | | |
+// /_/   \_\|_|   \___||_| |_||_||____/  \__|\___| \__,_||_| |_| |_||_|   \__,_||_|   |_| |_| |_|
+// 
+// Copyright 2015-2018 Łukasz "JustArchi" Domeradzki
+// Contact: JustArchi@JustArchi.net
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Markdig;
+using Markdig.Renderers;
+using Markdig.Syntax;
+using Markdig.Syntax.Inlines;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ArchiSteamFarm {
+	internal static class GitHub {
+		private static MarkdownDocument ExtractChangelogFromBody(string markdownText) {
+			if (string.IsNullOrEmpty(markdownText)) {
+				ASF.ArchiLogger.LogNullError(nameof(markdownText));
+				return null;
+			}
+
+			MarkdownDocument markdownDocument = Markdown.Parse(markdownText);
+			bool insideChangelog = false;
+
+			foreach (Block block in markdownDocument.ToList()) {
+				if (!insideChangelog) {
+					if (block is HeadingBlock headingBlock && (headingBlock.Inline.FirstChild != null) && headingBlock.Inline.FirstChild is LiteralInline literalInline && (literalInline.Content.ToString() == "Changelog")) {
+						insideChangelog = true;
+					}
+
+					markdownDocument.Remove(block);
+					continue;
+				}
+
+				if (block is ThematicBreakBlock) {
+					insideChangelog = false;
+					markdownDocument.Remove(block);
+				}
+			}
+
+			return markdownDocument;
+		}
+
+		internal static async Task<List<ReleaseResponse>> GetReleases(byte count) {
+			if (count == 0) {
+				ASF.ArchiLogger.LogNullError(nameof(count));
+				return null;
+			}
+
+			string releaseURL = SharedInfo.GithubReleaseURL + "?per_page=" + count;
+
+			return await GetReleasesFromURL(releaseURL).ConfigureAwait(false);
+		}
+
+		internal static async Task<ReleaseResponse> GetLatestRelease(bool stable = true) {
+			string releaseURL = SharedInfo.GithubReleaseURL + (stable ? "/latest" : "?per_page=1");
+
+			if (stable) {
+				return await GetReleaseFromURL(releaseURL).ConfigureAwait(false);
+			} else {
+				return (await GetReleasesFromURL(releaseURL).ConfigureAwait(false))?.ElementAt(0);
```

Comment:
Replace `ElementAt(0)` with `FirstOrDefault()`. You'll get exception when ASF returns non-null but empty `List`. You still need existing null check.

Reply:
Also it's a good idea to remove `else` since you have `return` above and do `var response = ` first before further null call.